src/quick-lint-js/error-tape.h

```diff
@@ -0,0 +1,65 @@
+// quick-lint-js finds bugs in JavaScript programs.
+// Copyright (C) 2020  Matthew Glazar
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef QUICK_LINT_JS_ERROR_DECORATOR_H
+#define QUICK_LINT_JS_ERROR_DECORATOR_H
+
+#include <quick-lint-js/error.h>
+#include <quick-lint-js/text-error-reporter.h>
+#include <quick-lint-js/vim-qflist-json-error-reporter.h>
+
+namespace quick_lint_js {
+class error_tape : public error_reporter {
+ public:
+  error_tape(error_reporter *reporter)
+    : reporter_(reporter),
+    error_ (false) {}
+
+  ~error_tape(){
+    delete reporter_;
+  }
```

Comment:
Almost-blocking: If the user of `error_tape` wants to transfer ownership of the `error_reporter` to the `error_tape`, the constructor should accept a `std::unique_ptr` instead of a raw pointer.

Blocking: Within `error_tape`, we should store the pointer in a `std::unique_ptr` instead of storing a raw pointer. A raw pointer is bug-prone here. There is a risk of double delete because `error_tape` has a copy constructor which copies the pointer.

Reply:
I think I will go with a template way so I will remove all about pointer into the code.